google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupAdvancedRemarketing.java

```diff
@@ -0,0 +1,225 @@
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    // Creates a rule targeting any user that visited the checkout page.
+    UserListRuleItemInfo checkoutRule =
+        UserListRuleItemInfo.newBuilder()
+            // The rule variable name must match a corresponding key name fired from a pixel.
+            // To learn more about setting up remarketing tags, visit
+            // https://support.google.com/google-ads/answer/2476688?hl=en.
+            // To learn more about remarketing events and parameters, visit
+            // https://support.google.com/google-ads/answer/7305793?hl=en.
+            .setName(StringValue.of("ecomm_pagetype"))
+            .setStringRuleItem(
+                UserListStringRuleItemInfo.newBuilder()
+                    .setOperator(UserListStringRuleItemOperator.EQUALS)
+                    .setValue(StringValue.of("checkout"))
+                    .build())
+            .build();
+
+    // Creates a rule targeting any user that had more than one item in their cart.
+    UserListRuleItemInfo cartSizeRule =
+        UserListRuleItemInfo.newBuilder()
+            // The rule variable name must match a corresponding key name fired from a pixel.
+            .setName(StringValue.of("cart_size"))
+            .setNumberRuleItem(
+                UserListNumberRuleItemInfo.newBuilder()
+                    .setOperator(UserListNumberRuleItemOperator.GREATER_THAN)
+                    .setValue(DoubleValue.of(1.0))
+                    .build())
+            .build();
+
+    // Creates a rule group that includes the checkout and cart size rules. Combining the two rule
+    // items into a UserListRuleItemGroupInfo object causes Google Ads to AND their rules together.
+    // To instead OR the rules together, each rule should be placed in its own rule item group.
+    UserListRuleItemGroupInfo checkoutAndCartSizeRuleGroup =
+        UserListRuleItemGroupInfo.newBuilder()
+            .addAllRuleItems(ImmutableList.of(checkoutRule, cartSizeRule))
+            .build();
+
+    // Creates the RuleItem for checkout start date.
+    // The tags and keys used below must have been in place in the past for the date range specified
+    // in the rules.
+    UserListRuleItemInfo startDateRule =
+        UserListRuleItemInfo.newBuilder()
+            // The rule variable name must match a corresponding key name fired from a pixel.
+            .setName(StringValue.of("checkoutdate"))
+            .setDateRuleItem(
+                UserListDateRuleItemInfo.newBuilder()
+                    .setOperator(UserListDateRuleItemOperator.AFTER)
+                    .setValue(StringValue.of("20191031"))
```

Comment:
Is there a FROM rule instead of an AFTER rule? It seems a bit odd to set "AFTER Oct 31" to achieve "FROM Nov 1". Same thoughts about Jan 1.

Reply:
No, these are the choices: https://developers.google.com/google-ads/api/reference/rpc/google.ads.googleads.v3.enums#google.ads.googleads.v3.enums.UserListDateRuleItemOperatorEnum.UserListDateRuleItemOperator

Reply:
Can you include this link as a reference for available values?